src/Nelmio/Alice/Instances/Instantiator/InstantiatorRegistry.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Alice package.
+ *  
+ * (c) Nelmio <hello@nelm.io>
+ *  
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Nelmio\Alice\Instances\Instantiator;
+
+use Nelmio\Alice\Exception\Instantiator\RuntimeException;
+use Nelmio\Alice\Fixtures\Fixture;
+use Nelmio\Alice\Instances\InstantiatorInterface;
+
+/**
+ * @author Théo FIDRY <theo.fidry@gmail.com>
+ */
+final class InstantiatorRegistry implements InstantiatorInterface
```

Comment:
rework of the old `Instantiator` class, there is little differences except that it's immutable